csrc/buffer.cpp

```diff
@@ -49,4 +50,53 @@ JByteArrayCritical::~JByteArrayCritical() { env_->ReleasePrimitiveArrayCritical(
 
 unsigned char* JByteArrayCritical::get() { return (unsigned char*)ptr_; }
 
+SimpleBuffer::SimpleBuffer(int size)
+    : buffer_(nullptr)
+{
+    buffer_ = (uint8_t*)malloc(size);
```

Comment:
I think we only have a SecureBuffer that allocates on stack. The scratch buffer size used when output clobbers input depends on the input/output buffers at runtime, so don't know the size before hand.
I don't think there is any issue in using malloc/free. The only concern could be that we do not zero memory after deallocation. The buffer is used to hold cipherText/plainText. In that sense it's as secure as how Java treats buffers by the garbage collector.
I'm open to suggestion to see if zeroing memory is needed before freeing up.

Reply:
Two things:
1. The code is updated so that before freeing the memory it is zeroed.
2. I see that we have [SecureAlloc](https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/csrc/env.h#L261) that could potentially be used. The main issue with this class is that it throws `std::bad_alloc`. In our JNI code, we only catch `java_ex`, which means any other type of exception could potentially result in undefined behavior in JNI calls.

Reply:
Replacing SecureAlloc with `OPENSSL_cleanse` makes sense. I wonder if we should extend that idea a little further, using `OPENSSL_malloc` and `OPENSSL_free` in SimpleBuffer. This would provide two benefits:

1. Automatically [call](https://github.com/aws/aws-lc/blob/e3c74c00a5303757e122087dfb6e9722e5dcb246/crypto/mem.c#L246) `OPENSSL_cleanse` on free
2. Automatically use the same underlying allocator as AWS-LC in case they ever decide to switch to e.g. [jemalloc](https://jemalloc.net/jemalloc.3.html) or something. due to portability concerns, IMO this is unlikely.

not a blocker at all, just perhaps something to consider.

Reply:
That's great suggestion. I changed it so that it will use OPENSS_{malloc,free}.